doc/modules/changes/20200506_haoyuanli

```diff
@@ -0,0 +1,3 @@
+Added: flexibility in input method for compositions and phases in the visco_plastic module inside material. New input method will allow single value for all compositions and phases or single value for all phases in a composition.
```

Comment:
This needs to be written in a broader sense, because your change will affect all uses of that function, not only the application you are thinking about at the moment. The following would be my suggestion, but feel free to adjust:
```suggestion
New: Added flexibility to the parse_map_to_double_array input method. The new input method will allow a single value for a given key (e.g. density of a composition), even if multiple values (e.g. for different phases) are expected. The single value will be expanded to be the same for all values associated with that key.
```